src/components/ImageUploader.js

```diff
@@ -0,0 +1,257 @@
+      this.setState({
+        invalidMessage: fileTooBig ? 'The file you selected is too big. The max file size is ' + this.props.maxFileSize + ' KB' : 'The file type is not supported'
+      });
+
+      this.props.onFileError(this.state.invalidMessage);
+    }
+  },
+
+  _handleRemoveFile (e) {
+    e.stopPropagation();
+
+    this.setState({
+      uploadedFile: null,
+      fileContent: null
+    });
+
+    this.props.onFileRemove();
+  },
+
+  _handleFileSelect (e) {
+    const file = e.target.files[0];
+
+    this._uploadFile(file);
+  },
+
+  _renderAllowedFileTypes () {
+    const lastIndex = this.props.allowedFileTypes.length - 1;
+
+    return this.props.allowedFileTypes.map((type, index) => {
+      const comma = lastIndex !== index ? ', ' : '';
+
+      return '.' + type + comma;
+    });
+  },
+
+  _renderDropzone () {
+    const styles = this.styles();
+
+    return (
+      <div>
+        <div style={styles.title}>Upload an Image:</div>
+        <div
+          onClick={this._onDropzoneClick}
+          onDragOver={this._onDragOver}
+          onDrop={this._onDrop}
+          style={styles.dropZoneWrapper}
+        >
+          <div style={[styles.dropZone, this.props.fileBeingDragged && styles.fileBeingDragged]}>
+            {this.props.fileBeingDragged ? (
+            <div style={styles.dropFileContainer}>
+              <Icon
+                size={32}
+                style={styles.dropFileIcon}
+                type='import'
+              />
+              <div style={styles.dropFileText}>Drop file to upload</div>
+            </div>) : (
+            <div style={styles.dropZoneText}>
+              Drag and drop files here or click to browse
+            </div>)}
+            {this.state.invalidMessage ? <div style={styles.invalidMessage}>{this.state.invalidMessage}</div> : null}
+          </div>
+          {this.state.fileContent ? (
+          <div style={styles.fileInfo}>
+            <img src={this.state.fileContent} />
+            <div style={styles.fileDetail}>
+              {this.state.uploadedFile.name}
+              ({numeral(this.state.uploadedFile.size / 1000).format('0.0')} KB)
+            </div>
+            <Button
+              onClick={this._handleRemoveFile}
+              primaryColor={this.props.color}
+              style={styles.removeButton}
+            >
+              Remove
+            </Button>
+          </div>) : null}
+          <input
+            name='files'
+            onChange={this._handleFileSelect}
+            ref='hiddenInput'
+            style={styles.hiddenInput}
+            type='file'
+          />
+        </div>
+        <div style={styles.fileDescription}>
+          File dimensions - 615 x 110 pixels <br/>
+          File types - {this._renderAllowedFileTypes()} <br/>
+          Max file size - <span>{this.props.maxFileSize}</span> KB
+        </div>
+      </div>
+    );
+  },
+
+  render () {
+    return (
+      <div>
+        {this._renderDropzone()}
```

Comment:
Pulling this into a separate method seems a bit overkill. I suggest eliminating `_renderDropzone` and moving it's content into the `render` method.